### PR TITLE
feat: LT - 1593 - F680 Clearance Types

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: applications/serializers/f680_clearance.py [hunk 1]
```diff
@@ -10,15 +10,26 @@
...
+    class Meta:
+        model = F680ClearanceType
+        fields = ("name",)
 
 
 class F680ClearanceViewSerializer(PartiesSerializerMixin, GenericApplicationViewSerializer):
     goods = GoodOnApplicationViewSerializer(many=True, read_only=True)
     destinations = serializers.SerializerMethodField()
     additional_documents = serializers.SerializerMethodField()
+    f680_clearance_types = F680ClearanceTypeSerializer(read_only=True, many=True)
```
Comment (jfaracik) [inline applications/serializers/f680_clearance.py]: Could this not just be `clearance_types`?

File: static/urls.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
     path("control-list-entries/", include("static.control_list_entries.urls")),
     path("private-venture-gradings/", include("static.private_venture_gradings.urls")),
     path("countries/", include("static.countries.urls")),
+    path("f680-clearance-types/", include("static.f680_clearance_types.urls")),
```
Comment (jfaracik) [inline static/urls.py]: Could this not just be `clearance-types`?

Comment (andytaylorkainos) [reply static/urls.py]: these types are only applicable to f680s so I think it makes sense here

File: applications/serializers/f680_clearance.py (hunk 1 above)
Comment (andytaylorkainos) [reply applications/serializers/f680_clearance.py]: From the perspective of an F680 applicaton it ofcourse doesnt make sense to duplicate the `f680_clearance` information in the name if we already know.

However, we refer to our cases and applications from `case` or `application` where it makes more sense to have this information in the name as it only relates to cases/applications that are F680 instances:
`case.application.f680_clearance_details`

I honestly don't mind what we name it, but i dont think it's worth the effort at this point for a system-wide change (seeding to DB, updating all request and response data on two frontends and updating the ui/shared seeding scripts)

Comment (cstobart) [inline applications/serializers/f680_clearance.py]: duplication of class name as attribute, please change

File: static/urls.py (hunk 2 above)
Comment (jfaracik) [reply static/urls.py]: What about `f680/clearance-types/`

File: applications/creators.py [hunk 3]
```diff
@@ -232,6 +232,9 @@ def _validate_f680_clearance(draft, errors):
     if SiteOnApplication.objects.filter(application=draft).exists():
         errors["location"] = strings.Applications.F680.LOCATIONS
 
+    if not draft.f680_clearance_types.exists():
+        errors["f680-details"] = "Cannot create an application without F680 details"
```
Comment (cstobart) [inline applications/creators.py]: Will people know what specifically constitutes a F680 detail?

File: applications/models.py [hunk 4]
```diff
@@ -163,7 +164,7 @@ class GiftingClearanceApplication(BaseApplication):
 
 # F680 includes End User & Third parties
 class F680ClearanceApplication(BaseApplication):
-    pass
+    f680_clearance_types = models.ManyToManyField(F680ClearanceType, related_name="f680_clearance_type_application")
```
Comment (cstobart) [inline applications/models.py]: duplication of class name as attribute

File: applications/creators.py (hunk 3 above)
Comment (andytaylorkainos) [reply applications/creators.py]: i think this is just to match the section on the application-overview (which content has named as f680-details. good point tho, will rename to match the fields required)

Comment (cstobart) [reply applications/creators.py]: 👍

File: applications/serializers/f680_clearance.py (hunk 1 above)
Comment (andytaylorkainos) [reply applications/serializers/f680_clearance.py]: Damn.

File: applications/models.py (hunk 4 above)
Comment (andytaylorkainos) [reply applications/models.py]: Done

File: applications/serializers/f680_clearance.py (hunk 1 above)
Comment (andytaylorkainos) [reply applications/serializers/f680_clearance.py]: Done

Comment (andytaylorkainos) [reply applications/serializers/f680_clearance.py]: Looks like others agreed with you, done! :)

**Pull request merged**